conans/client/remote_manager.py

```diff
@@ -129,6 +130,9 @@ def get_package(self, pref, dest_folder, remote, output, recorder):
         t1 = time.time()
         try:
             pref = self._resolve_latest_pref(pref, remote)
+            snapshot = self._call_remote(remote, "get_package_snapshot", pref)
+            if [CONANINFO, CONAN_MANIFEST, PACKAGE_TGZ_NAME] != snapshot:
```

Comment:
That check is too much restrictive, we don't want to check here that the snapshot is exactly that, but we have all we need. I would check if all the files are there (and checking that the tgz could have a different name)